### DISCUSSION TITLE
[Feature] A denser, more focused representation of the tasks in the list (configuration possiblity)

### DISCUSSION THREAD

**Discussion by DoS007** [TechbeeAT/jtxBoard#1241] [Ideas]
At the moment the most you can do is compact and (in german) "flach".

First is jtx board. And second is an example of tasks.org ( there you can click e.g. on a specific tasklist, but it's not only that, look at the space what is possible). Much cleaner in my view.
<p align="center">
<img src="https://github.com/TechbeeAT/jtxBoard/assets/11618958/abc9a84c-c93c-4d89-8f95-90fd7c03c918" width=200>

<img src="https://github.com/TechbeeAT/jtxBoard/assets/11618958/b25dd8ca-12fd-4bf3-a16d-ba504ebb6d49" width=200>
</p>

**Comment by patrickunterwegs** [comment]
Hi @DoS007 , the problem are accessibility and design guidelines here. I'm using the UI elements from Material 3 (provided by Google). Those automatically have certain padding values to ensure a minimum tap-size. You can see this e.g. when you add a note with text only, then the row is much smaller. But when you add a task, you can see that the whole row gets higher, because the checkbox takes so much space around. 
The Tasks app seems to ignore all those accessibility guidelines. 
So basically it would be possible, but first I need to explicitly work around the guidelines, and then it could also lead to frustration of users, who can't tap properly anymore on the checkbox because the clickable area gets too small...

**Reply by DoS007** [reply]
@patrickunterwegs Hi, thanks for your promptly response and that explanation. I personally prefer it like in second picture because it's cleaner in my view. 

>"and then it could also lead to frustration of users, who can't tap properly anymore on the checkbox because the clickable area gets too small..."

It could be set as an *option* like it also is on tasks.org. The shown second screenshot isn't default padding.

**Reply by patrickunterwegs** [reply]
Hi @DoS007 , I have tried to make it denser now, I removed the "badges" and I'd just show a text line now with the meta info. I think it's much clearer now and saves a lot of space. What do you think?

![image](https://github.com/TechbeeAT/jtxBoard/assets/72232737/701ba127-0e1b-4af9-86df-6bd83ba575d1)

**Reply by DoS007** [reply]
Hi, @patrickunterwegs , thanks for your implementation. I cannot see the image because when i click i get a failure. ?

**Reply by patrickunterwegs** [reply]
Hi @DoS007 , sorry, I'm not sure why images sometimes just don't work. 
Can you try this link?
[Image](https://private-user-images.githubusercontent.com/72232737/318373389-701ba127-0e1b-4af9-86df-6bd83ba575d1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE5OTUzOTgsIm5iZiI6MTcxMTk5NTA5OCwicGF0aCI6Ii83MjIzMjczNy8zMTgzNzMzODktNzAxYmExMjctMGUxYi00YWY5LTg2ZGYtNmJkODNiYTU3NWQxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDAxVDE4MTEzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA5ZTdkNjBmZWFhNTE5NWVkNTBjNGNhMGEyYjY5OTBhODcwNGU4YmU0ZTU4MmFjZGU2Njk0ZWQwMWQyYzg3ZmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.iUfdVXVLrf7e8ytik24VnALv3UqfSxC8h3Xz_5AcFuY)

Or maybe now it just works: 

![318373389-701ba127-0e1b-4af9-86df-6bd83ba575d1](https://github.com/TechbeeAT/jtxBoard/assets/72232737/09a1173b-91cd-4953-acae-fbb6ac85ece4)

**Reply by DoS007** [reply]
Hi @patrickunterwegs both links don't work either. Even tried another browser. Maybe upload to cloud and share link?

**Reply by patrickunterwegs** [reply]
Sorry @DoS007 , I don't know why this happens, the images are still shown for me. 
Can you try this link: https://postimg.cc/D8pMz50H

**Reply by DoS007** [reply]
@patrickunterwegs Yeah,  that worked. Probably the other uploads in some space of you which you can only view with your login or so.

I think it's more focused in that view because now the task-names itself (and that's important) are more pronounced this way.

What i give thought is: Couldn't the Meta info be removed by e.g. giving specific views (a) e.g. only view tasks of one specific lists or b) give a general view setting in which one can select wich "badges" or meta-info to show [hashtags e.g.]) or outsource it to e.g. a c) color or use a d) reduced simple symbol in the same line with the task-name text.

**Reply by Ceaus** [reply]
I've raised a ticket for a similar issue a while back. And I've received the exact same reply ("design guidelines"). Personally I'd argue the guidelines are wrong. Or not fitting for this particular situation. This is now becoming form before function. And as soon as the UI/UX is getting a hit, I think it's time to count your pennies.

**Reply by patrickunterwegs** [reply]
Thanks @DoS007 for your feedback!
There is an open issue where I wanted to make the list view more flexible anyway, but this is for the future. For now I will replace badges in the meta-info with just text and symbols 👍 

@Ceaus Well, some design guidelines do have their reasons. But I agree with you, that Google overdid it in the Material 3 guidelines and force you to waste a lot of space. I have some ideas how to make it better, but this will take some time.

**Comment by leaumar** [comment]
I've been using tasks.org for a long time and decided to try jtx instead, because I have a new device but setting up tasks.org again is a lot of work (because getting the gui the way I like it takes a lot of tedious settings for which there's no [export/import](https://github.com/tasks/tasks/issues/951) yet). I then bought pro after a few minutes, thinking the premium "compact" mode would be... _compact_

I understand the argument about the design language being what it is, but I at least expected to be able to remove the clutter elements around my tasks, i.e. the collection name, status, and priority badges. I don't want any of them, just labeled checkboxes under group headings (and caldav, to be clear). If it's possible, I'm not seeing the relevant settings.

Thanks for your work on this app, as a dev myself I'm aware there's no "making demands" here. But my review as a user is that this is not an efficient gui, fitting merely 5 tasks on my entire screen at a time, and I'll keep using tasks.org (which comfortably manages 12).

If only we didn't have to beg every app's devs to add settings export/import too... :(

**Reply by Ceaus** [reply]
As a matter of fact, I am a recurring donor, as I want to support the developer for the  software I'm using. My primary reason for using jtxBoard are the hierarchical tasks and the notes which can be synced via davcal over DAVx5. I agree jtxBoard seems to be over-engineered leading to a somewhat cluttered UI. The focus for usability and UX seems to be a little down, which could be the reason. Regardless, I appreciate the work @patrickunterwegs is doing and I'll keep supporting him for the foreseeable future.

**Comment by fossy1** [comment]
Is there some way to collapse subtasks into the parent task, when we just went to see top level tasks (preferably with an indicator of how many subtasks each parent task has)?

**Comment by bvdbos** [comment]
A lot of space is wasted indeed. Please remove the tags. The horizontal line shouldn't appear between tasks but should above the task-category.
<img width="1080" height="2340" alt="afbeelding" src="https://github.com/user-attachments/assets/d92eb5b5-4150-4e7a-bbcc-89466083fc23" />
